sway/commands/bar.c

```diff
@@ -0,0 +1,57 @@
+#include <string.h>
+#include <strings.h>
+#include <wlr/util/log.h>
+#include "sway/commands.h"
+#include "sway/config.h"
+#include "util.h"
+
+struct cmd_results *cmd_bar(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "bar", EXPECTED_AT_LEAST, 1))) {
+		return error;
+	}
+
+	if (config->reading && strcmp("{", argv[0]) != 0) {
+		return cmd_results_new(CMD_INVALID, "bar",
+				"Expected '{' at start of bar config definition.");
+	}
+
+	if (!config->reading) {
+		if (argc > 1) {
+			if (strcasecmp("mode", argv[0]) == 0) {
+				return bar_cmd_mode(argc-1, argv + 1);
+			}
+
+			if (strcasecmp("hidden_state", argv[0]) == 0) {
+				return bar_cmd_hidden_state(argc-1, argv + 1);
+			}
+		}
+		return cmd_results_new(CMD_FAILURE, "bar", "Can only be used in config file.");
+	}
+
+	// Create new bar with default values
+	struct bar_config *bar = default_bar_config();
+	if (!bar) {
+		return cmd_results_new(CMD_FAILURE, "bar", "Unable to allocate bar state");
+	}
+
+	// set bar id
+	int i;
+	for (i = 0; i < config->bars->length; ++i) {
+		if (bar == config->bars->items[i]) {
+			const int len = 5 + numlen(i); // "bar-" + i + \0
+			bar->id = malloc(len * sizeof(char));
+			if (bar->id) {
+				snprintf(bar->id, len, "bar-%d", i);
+			} else {
+				return cmd_results_new(CMD_FAILURE, "bar", "Unable to allocate bar ID");
```

Comment:
`cmd_results_new` allocates stuff

Reply:
If you have a crazy bar ID though that could fail and this might not